changelog 2.20.0 (#8124)
timescale/timescaledb: CHANGELOG.md

* *Continuous aggregates*  now include experimental support for *window functions and non-immutable functions*, extending the analytics use cases they can solve.
* Several quality-of-life improvements have been introduced: job names for continuous aggregates are now more descriptive, you can assign custom names to them, and it is now possible to add unique constraints along with `ADD COLUMN` operations in the columnstore.
* Improved management and optimization of chunks with the ability to split large uncompressed chunks at a specified point in time using the `split_chunk` function. This new function complements the existing `merge_chunk` function that can be used to merge two small chunks into one larger chunk. 
* Enhancements to the default behavior of the columnstore now provide better *automatic assessments* of `segment by` and `order by` columns, reducing the need for manual configuration and simplifying initial setup.

**PostgreSQL 14 support removal announcement**

Following the deprecation announcement for PostgreSQL 14 in TimescaleDB v2.19.0, PostgreSQL 14 is no longer supported in TimescaleDB v2.20.0. The currently supported PostgreSQL major versions are 15, 16, and 17.

> 

Author: ```suggestion

**Windows: Postgres support**

This TimescaleDB release supports the Postgres versions `15.12.0`, `16.8.0` and `17.4.0` on Windows.

```

* [#7762](https://github.com/timescale/timescaledb/pull/7762) Speed up the queries that use minmax sparse indexes on compressed tables by changing the index TOAST storage type to `MAIN`. This applies to newly compressed chunks
* [#7785](https://github.com/timescale/timescaledb/pull/7785) Do `DELETE` instead of `TRUNCATE` when locks aren't acquired
* [#7852](https://github.com/timescale/timescaledb/pull/7852) Allow creating foreign key constraints on compressed tables
* [#7854](https://github.com/timescale/timescaledb/pull/7854) Remove support for PG14
* [#7864](https://github.com/timescale/timescaledb/pull/7854) Allow adding CHECK constraints to compressed chunks
* [#7868](https://github.com/timescale/timescaledb/pull/7868) Allow adding columns with `CHECK` constraints to compressed chunks
* [#7874](https://github.com/timescale/timescaledb/pull/7874) Support for SkipScan for distinct aggregates over the same column
* [#7877](https://github.com/timescale/timescaledb/pull/7877) Remove blocker for unique constraints with `ADD COLUMN`
* [#7878](https://github.com/timescale/timescaledb/pull/7878) Don't block non-immutable functions in continuous aggregates

> * [#7880](https://github.com/timescale/timescaledb/pull/7880) Add experimental support for window functions in caggs

Reviewer: ```suggestion
* [#7880](https://github.com/timescale/timescaledb/pull/7880) Add experimental support for window functions in continuous aggregates
```

* [#7899](https://github.com/timescale/timescaledb/pull/7899) Vectorized decompression and filtering for boolean columns

> * [#7915](https://github.com/timescale/timescaledb/pull/7915) New option `refresh_newest_first` to CAgg refresh policy API

Reviewer: ```suggestion
* [#7915](https://github.com/timescale/timescaledb/pull/7915) New option `refresh_newest_first` to continuous aggregate refresh policy API
```

* [#8115](https://github.com/timescale/timescaledb/pull/8115) Add batch size limiting during compression

**Bugfixes**
* [#7862](https://github.com/timescale/timescaledb/pull/7862) Release cache pin when checking for `NOT NULL`
* [#7909](https://github.com/timescale/timescaledb/pull/7909) Update compression stats when merging chunks
* [#7928](https://github.com/timescale/timescaledb/pull/7928) Don't create a hypertable for implicitly published tables
* [#7982](https://github.com/timescale/timescaledb/pull/7982) Fix crash in batch sort merge over eligible expressions
* [#8008](https://github.com/timescale/timescaledb/pull/8008) Fix compression policy error message that shows number of successes
* [#8031](https://github.com/timescale/timescaledb/pull/8031) Fix reporting of deleted tuples for direct batch delete
* [#8033](https://github.com/timescale/timescaledb/pull/8033) Skip default `segmentby` if `orderby` is explicitly set
* [#8061](https://github.com/timescale/timescaledb/pull/8061) Ensure settings for a compressed relation are found
* [#7515](https://github.com/timescale/timescaledb/pull/7515) Add missing lock to Constraint-aware append

> * [#8067](https://github.com/timescale/timescaledb/pull/8067) Make sure Hypercore TAM parent is vacuumed

Reviewer: ```suggestion
* [#8067](https://github.com/timescale/timescaledb/pull/8067) Make sure hypercore TAM parent is vacuumed
```

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

## 2.20.0 (2025-05-15)

> This release contains performance improvements and bug fixes since the 2.18.2 release. We recommend that you upgrade at the next available opportunity.

Reviewer 2: ```suggestion
This release contains performance improvements and bug fixes since the 2.19.3 release. We recommend that you upgrade at the next available opportunity.
```

* Several quality-of-life improvements have been introduced: job names for continuous aggregates are now more descriptive, you can assign custom names to them, and it is now possible to add unique constraints along with `ADD COLUMN` operations in the columnstore.
* Improved management and optimization of chunks with the ability to split large uncompressed chunks at a specified point in time using the `split_chunk` function. This new function complements the existing `merge_chunk` function that can be used to merge two small chunks into one larger chunk. 
* Enhancements to the default behavior of the columnstore now provide better *automatic assessments* of `segment by` and `order by` columns, reducing the need for manual configuration and simplifying initial setup.

**PostgreSQL 14 support removal announcement**

Following the deprecation announcement for PostgreSQL 14 in TimescaleDB v2.19.0, PostgreSQL 14 is no longer supported in TimescaleDB v2.20.0. The currently supported PostgreSQL major versions are 15, 16, and 17.

**Windows: Postgres support**

> This TimescaleDB release supports the Postgres versions `15.12.0`, `16.8.0` and `17.4.0` on Windows.

Reviewer 3: I would leave this out. We couldnt test 17.5 yet because they are not yet in chocolatey but we dont expect them to not work. The `.0` suffix is chocolatey specific  so we should probably drop it if we mention this.

> This TimescaleDB release supports the Postgres versions `15.12.0`, `16.8.0` and `17.4.0` on Windows.

Reviewer 2: ```suggestion
This TimescaleDB release supports Postgres versions `15.12.0`, `16.8.0` and `17.4.0` on Windows.
```
